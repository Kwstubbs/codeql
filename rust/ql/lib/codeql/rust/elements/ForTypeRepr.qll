// generated by codegen, do not edit
/**
 * This module provides the public class `ForTypeRepr`.
 */

private import internal.ForTypeReprImpl
import codeql.rust.elements.ForBinder
import codeql.rust.elements.TypeRepr

/**
 * A type with a higher-ranked `for` modifier. This is currently not valid Rust syntax (`for<...>` can
 * only be applied to traits to form a `TypeBound`).
 *
 * For example:
 * ```rust
 * fn foo(value: for<'a> usize) {}  // DOESN'T COMPILE
 * //            ^^^^^^^^^^^^^
 * ```
 */
final class ForTypeRepr = Impl::ForTypeRepr;
