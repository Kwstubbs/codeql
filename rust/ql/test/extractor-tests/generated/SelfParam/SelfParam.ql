// generated by codegen, do not edit
import codeql.rust.elements
import TestUtils

query predicate instances(
  SelfParam x, string getNumberOfAttrs__label, int getNumberOfAttrs, string hasTypeRepr__label,
  string hasTypeRepr, string isRef__label, string isRef, string isMut__label, string isMut,
  string hasLifetime__label, string hasLifetime, string hasName__label, string hasName
) {
  toBeTested(x) and
  not x.isUnknown() and
  getNumberOfAttrs__label = "getNumberOfAttrs:" and
  getNumberOfAttrs = x.getNumberOfAttrs() and
  hasTypeRepr__label = "hasTypeRepr:" and
  (if x.hasTypeRepr() then hasTypeRepr = "yes" else hasTypeRepr = "no") and
  isRef__label = "isRef:" and
  (if x.isRef() then isRef = "yes" else isRef = "no") and
  isMut__label = "isMut:" and
  (if x.isMut() then isMut = "yes" else isMut = "no") and
  hasLifetime__label = "hasLifetime:" and
  (if x.hasLifetime() then hasLifetime = "yes" else hasLifetime = "no") and
  hasName__label = "hasName:" and
  if x.hasName() then hasName = "yes" else hasName = "no"
}

query predicate getAttr(SelfParam x, int index, Attr getAttr) {
  toBeTested(x) and not x.isUnknown() and getAttr = x.getAttr(index)
}

query predicate getTypeRepr(SelfParam x, TypeRepr getTypeRepr) {
  toBeTested(x) and not x.isUnknown() and getTypeRepr = x.getTypeRepr()
}

query predicate getLifetime(SelfParam x, Lifetime getLifetime) {
  toBeTested(x) and not x.isUnknown() and getLifetime = x.getLifetime()
}

query predicate getName(SelfParam x, Name getName) {
  toBeTested(x) and not x.isUnknown() and getName = x.getName()
}
