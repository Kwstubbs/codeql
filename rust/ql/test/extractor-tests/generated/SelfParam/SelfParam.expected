instances
| gen_self_param.rs:6:10:6:14 | SelfParam | getNumberOfAttrs: | 0 | hasTypeRepr: | no | isRef: | yes | isMut: | no | hasLifetime: | no | hasName: | yes |
| gen_self_param.rs:7:10:7:18 | SelfParam | getNumberOfAttrs: | 0 | hasTypeRepr: | no | isRef: | yes | isMut: | yes | hasLifetime: | no | hasName: | yes |
| gen_self_param.rs:8:12:8:15 | SelfParam | getNumberOfAttrs: | 0 | hasTypeRepr: | no | isRef: | no | isMut: | no | hasLifetime: | no | hasName: | yes |
| gen_self_param.rs:9:11:9:18 | SelfParam | getNumberOfAttrs: | 0 | hasTypeRepr: | no | isRef: | no | isMut: | yes | hasLifetime: | no | hasName: | yes |
| gen_self_param.rs:10:15:10:22 | SelfParam | getNumberOfAttrs: | 0 | hasTypeRepr: | no | isRef: | yes | isMut: | no | hasLifetime: | yes | hasName: | yes |
getAttr
getTypeRepr
getLifetime
| gen_self_param.rs:10:15:10:22 | SelfParam | gen_self_param.rs:10:16:10:17 | 'a |
getName
| gen_self_param.rs:6:10:6:14 | SelfParam | gen_self_param.rs:6:11:6:14 | self |
| gen_self_param.rs:7:10:7:18 | SelfParam | gen_self_param.rs:7:15:7:18 | self |
| gen_self_param.rs:8:12:8:15 | SelfParam | gen_self_param.rs:8:12:8:15 | self |
| gen_self_param.rs:9:11:9:18 | SelfParam | gen_self_param.rs:9:15:9:18 | self |
| gen_self_param.rs:10:15:10:22 | SelfParam | gen_self_param.rs:10:19:10:22 | self |
